### PR TITLE
imap/xapian.cpp: more C++ tweaks

### PR DESCRIPTION
**Pull request by dilyanpalauzov**
* xapian_dbw’s members stemmer and stopper are removed.  The former was not used.  The value of the latter was always nullptr and this value is passed to dbw→term_generator.set_stopper().

* set the visibility to "hidden" to internal symbols

* remove the word “struct” for defining a type, as it is not necessary in C++.

### CODE REVIEW

File: imap/xapian_wrap.cpp [hunk 1]
```diff
@@ -417,82 +415,73 @@ void xapian_check_if_needs_reindex(const strarray_t *sources, strarray_t *torein
 
 struct xapian_dbw
 {
-    Xapian::WritableDatabase *database;
-    Xapian::Stem *stemmer;
-    Xapian::TermGenerator *term_generator;
-    Xapian::Document *document;
+    Xapian::WritableDatabase database;
```
Comment (rsto) [inline imap/xapian_wrap.cpp]: I remember we ran into trouble passing C++ objects in this struct to the C code. That's why we used pointers. But I can't replicate that issue with my current setup, so probably it's due to and old compiler version we are using. I'll check.

**Comment by elliefm** [comment]
@rsto You reviewed this in 2019 and requested changes.  Looks like Dilyan made some changes, but you didn't come back and re-review.  Can you have a look again please?

@dilyanpalauzov Can you rebase this on current master and fix the conflicts?  Or just close it if it's no longer useful.  Thanks

**Comment by dilyanpalauzov** [comment]
I think these changes are good, they switch from pointers to C++ objects; utilize C++ syntax for methods with `try {…} catch {…}` being the sole function body, without surrounding `{}`.

If I update the changes now, so that these can be applied clearly towards the current master branch, it is in no way evident, whether in five years I will be asked again to refresh the snippets, so that they can be applied without conflicts to the master branch.

So I think it would be good if somebody looks at the proposed changes, and adjusts them to the current tip of git, but I am personally not going to do this.

**Comment by rsto** [comment]
> So I think it would be good if somebody looks at the proposed changes, and adjusts them to the current tip of git, but I am personally not going to do this.

@dilyanpalauzov I am sorry that I had first asked for changes, then didn't follow up. I should have asked from the start to dissect this commit into smaller units of change, e.g. one for each stylistic or idiomatic update, which would make it simpler to reason about. I won't ask you do that now but will rather look at the current PR as-is and then decide how to proceed with it.

**Comment by rsto** [comment]
I'm inclined to close this pull request. Again, my apologies for having it linger around so long. I think we should approach this with more structure and smaller changes and I have started this effort with https://github.com/cyrusimap/cyrus-imapd/pull/5017 . I think that we should start using clang-tidy as a common base to define what "sane C++" looks like and I appreciate any additional PRs or contributions that follow this direction.

**Pull request closed without merging**